src/plugins/data/server/search/search_service.ts

```diff
@@ -254,8 +254,7 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
           const searchSourceDependencies: SearchSourceDependencies = {
             getConfig: <T = any>(key: string): T => uiSettingsCache[key],
             search: asScoped(request).search,
-            // onResponse isn't used on the server, so we just return the original value
-            onResponse: (req, res) => res,
+            onResponse: (req, res) => shimHitsTotal(res),
```

Comment:
I discovered that `shimHitsTotal` is applied in the low-level search _routes_, but not in the actual service that's available on the plugin contract, so we need to apply it here for search source.

(This is only necessary on the server, as the client is going through the http routes)

Reply:
cc @lizozom @lukasolson Whenever `shimHitsTotal` is updated to be applied at the search strategy level instead of the route level, this change can be reverted.